don't run deleter if it is null. issue #43166

bdhirsh opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#43271 don't run deleter if it is null. issue #43166**

Differential Revision: [D23301090](https://our.internmc.facebook.com/intern/diff/D23301090)

bdhirsh committed:
don't run deleter if it is null. issue #43166

[ghstack-poisoned]

ezyang commented on aten/src/ATen/test/dlconvertor_test.cpp:
```diff
@@ -30,3 +30,15 @@ TEST(TestDlconvertor, TestDlconvertorNoStrides) {
+
+  Tensor a = rand({3, 4});
+  DLManagedTensor* dlMTensor = toDLPack(a);
+  dlMTensor->deleter = nullptr;
```
This doesn't look right. What you have written here effectively says, "Allocate a DLPack tensor from PyTorch, and then *leak* the memory associated with it by forgetting the deleter." While I agree it exercises the code, this isn't code you'd ever want to actually see in production. Your best bet is probably to manually allocate a DLPack tensor with no deleter; you can look at toDLPack for an idea of what fields you need to fill.

bdhirsh committed:
Update on "don't run deleter if it is null. issue #43166"




[ghstack-poisoned]

ezyang commented on aten/src/ATen/test/dlconvertor_test.cpp:
```diff
@@ -8,6 +8,7 @@
 #include <sstream>
 
 using namespace at;
+
```
nit: unnecessary change

bdhirsh committed:
Update on "don't run deleter if it is null. issue #43166"




[ghstack-poisoned]

bdhirsh committed:
Update on "don't run deleter if it is null. issue #43166"


Differential Revision: [D23301090](https://our.internmc.facebook.com/intern/diff/D23301090)

[ghstack-poisoned]
